loader/integration.go

```diff
@@ -55,3 +56,31 @@ func GetLatestPipelineRunForSnapshotAndScenario(adapterClient client.Client, ctx
 
 	return nil, err
 }
+
+// RemoveFinalizerFromAllIntegrationPipelineRunsOfSnapshot fetches all the Integration
+// PipelineRuns associated with the given Snapshot and each of the IntegrationTestScenarios.
+// After fetching them, it removes the finalizer from the PipelineRun, and returns error if any.
+func RemoveFinalizerFromAllIntegrationPipelineRunsOfSnapshot(adapterClient client.Client, logger helpers.IntegrationLogger, ctx context.Context, snapshot applicationapiv1alpha1.Snapshot, finalizer string) error {
+	integrationPipelineRuns := &tektonv1beta1.PipelineRunList{}
+	opts := []client.ListOption{
+		client.InNamespace(snapshot.Namespace),
+		client.MatchingLabels{
+			"appstudio.openshift.io/snapshot": snapshot.Name,
+		},
+	}
+
+	err := adapterClient.List(ctx, integrationPipelineRuns, opts...)
+	if err != nil {
+		return err
+	}
+
+	// Remove finalizer from each of the PipelineRuns
+	for _, pipelineRun := range integrationPipelineRuns.Items {
+		err = helpers.RemoveFinalizerFromPipelineRun(adapterClient, logger, ctx, &pipelineRun, finalizer)
```

Comment:
```suggestion
	for _, pipelineRun := range integrationPipelineRuns.Items {
	        pipelineRun := pipelineRun
		err = helpers.RemoveFinalizerFromPipelineRun(adapterClient, logger, ctx, &pipelineRun, finalizer)
```

```
[/github/workspace/loader/integration.go:79] - G601 (CWE-118): Implicit memory aliasing in for loop. (Confidence: MEDIUM, Severity: MEDIUM)
    78: 	for _, pipelineRun := range integrationPipelineRuns.Items {
  > 79: 		err = helpers.RemoveFinalizerFromPipelineRun(adapterClient, logger, ctx, &pipelineRun, finalizer)
    80: 		if err != nil 
```

Reply:
Thank you, updated. I'll keep this in mind.